swap_meet/vendor.py

```diff
@@ -1,2 +1,117 @@
+'''
+Vendor class represents individuals trading at swap meet.
+Includes the attribute 'inventory', which is a list of Item instances.
+Includes numerous methods to work with and modify vendors' inventories.
+'''
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        '''
+        Adds one item to vendor's inventory and returns item. 
+        Helper function to swap_items method.
+        '''
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        '''
+        Removes one item from vendor's inventory and returns item. 
+        Returns False if item not found in inventory. 
+        Helper function to swap_items method.
+        '''
+        try:
+            self.inventory.remove(item)
+            return item
+        except ValueError:
+            return False
+
+    def get_by_category(self, category):
+        '''
+        Takes a category as parameter and returns instance's inventory list filtered to only items belonging to that category.
+        Returns empty list if none found. 
+        Helper function to get_best_by_category method.
+        '''
+        return [item for item in self.inventory if item.category == category]
+
+    def swap_items(self, other, item_given, item_received):
+        '''
+        Receives parameters of other vendor instance, item to give that instance and item to receive from that instance. 
+        Swaps two items between the vendor instance's inventory and other vendor instance's inventory. 
+        Returns True if successful and False if either item is not found in inventory. 
+        Helper function to several other methods.
+        '''
+        if item_given not in self.inventory or item_received not in other.inventory:
+            return False
+
+        self.remove(item_given)
+        self.add(item_received)
+        other.remove(item_received)
+        other.add(item_given)
+        return True
+
+    def swap_first_item(self, other):
+        '''
+        Receives other vendor instance as parameter. 
+        Trades the first item listed in vendor instance's inventory with the first item listed in other vendor instance's inventory. 
+        Returns True if successful and False if either inventory is falsey.
+        '''
+        if not self.inventory or not other.inventory:
+            return False
+        return self.swap_items(other, self.inventory[0], other.inventory[0])
```

Comment:
👍

Reply:
Currently, swapping any items has a time complexity of O(n) due to the `remove` method in `swap_items`. We can make a subtle improvement to O(1) by using multiple-assignment instead:

```
        self.inventory[0], vendor.inventory[0] = vendor.inventory[0], self.inventory[0]
```        

Here's more info on Python multiple-assignment: https://www.w3schools.com/python/gloss_python_assign_value_to_multiple_variables.asp